File: recursive.cpp
#include <iostream>

using namespace std;

int factorial(int number);

int main()
{
    int number, result;
    cout << "Enter a number: \n";
    cin >> number;
    result = factorial(number);
    cout << result;
    return 0;
}

int factorial(int number)
{
    if(number <= 1) {
        return 1;
    } else {
        return number * factorial(number - 1);
    }
}

File: recursive.kt
import java.util.Scanner

fun factorial(n: Int): Int {
    if (n < 1) {
      return 1
    } else {
      return n * factorial(n - 1)
    }
}

fun main() {
    val reader = Scanner(System.`in`)
    print("Number:")
    var input: Int = reader.nextInt()
    var result: Int = factorial(input)
    println("$result")
}
